Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,124 @@
+            destinations.append(destination)
+        }
+    }
+
+    public class func unregister(_ destination: LogDestination) {
+        destinations = destinations.filter({ registeredDestinaton -> Bool in
+            return registeredDestinaton.instanceId != destination.instanceId
+        })
+    }
+
+    public class func removeAllDestinations() {
+        destinations.removeAll()
+    }
+
+    // MARK:- Logging
+
+    public class func verbose(_ message: @autoclosure () -> String,
+                              file: String = #file,
+                              function: String = #function,
+                              line: Int = #line) {
+
+        log(level: .verbose, message: message, file: file, function: function, line: line)
+    }
+
+    public class func debug(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line) {
+
+        log(level: .debug, message: message, file: file, function: function, line: line)
+    }
+
+    public class func info(_ message: @autoclosure () -> String,
+                           file: String = #file,
+                           function: String = #function,
+                           line: Int = #line) {
+
+        log(level: .info, message: message, file: file, function: function, line: line)
+    }
+
+    public class func warning(_ message: @autoclosure () -> String,
+                              file: String = #file,
+                              function: String = #function,
+                              line: Int = #line) {
+
+        log(level: .warning, message: message, file: file, function: function, line: line)
+    }
+
+    public class func error(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line) {
+
+        log(level: .error, message: message, file: file, function: function, line: line)
+    }
+
+    public class func log(level: Level,
+                          message: @autoclosure () -> String,
+                          file: String = #file,
+                          function: String = #function,
+                          line: Int = #line) {
+
+        let item = Log.Item(level: level, message: message(), file: file,
+                            thread: threadName(), function: function, line: line)
+
+        for destination in destinations {
+            if itemShouldBeLogged(destination: destination, item: item) {
+                destination.write(item: item)
+            }
+        }
+    }
+
+    // MARK:- Private Methods
+
+    private class func threadName() -> String {
+
+        if Thread.isMainThread {
+            return "main-thread"
+        }
+        else {
+            if let threadName = Thread.current.name, !threadName.isEmpty {
```

Comment:
Now that I look at this with fresh eyes, this logic could be moved into a `Thread` extension, don't you think?